### PR TITLE
fixed issue #159

### PR DESCRIPTION
**Pull request by UlrichHaboeck75**
This is the fix for #159

### CODE REVIEW

**Review by 95DDB — APPROVED** [review]
Okey from code perspective

File: r1cs/gadgets/std/src/fields/nonnative/nonnative_field_gadget.rs [hunk 1]
```diff
@@ -1126,8 +1126,8 @@ impl<SimulationF: PrimeField, ConstraintF: PrimeField> FieldGadget<SimulationF,
     // ``
```
Comment (DDT92) [inline r1cs/gadgets/std/src/fields/nonnative/nonnative_field_gadget.rs:1111]: we shuld have (num_groups -1) * ...

File: r1cs/gadgets/std/src/fields/nonnative/nonnative_field_gadget.rs [hunk 2]
```diff
@@ -1126,8 +1126,8 @@ impl<SimulationF: PrimeField, ConstraintF: PrimeField> FieldGadget<SimulationF,
...
     // constraints, where 
     // ``
-    //      surfeit' =  log(num_limbs * (num_adds(prod) + 1) + 1)
-    //              = log(num_limbs^2 * (num_add(L)+1) * (num_add(R) + 1)),
+    //      surfeit' =  log(num_limbs + 2 * (num_adds(prod) + 1))
+    //              = log(num_limbs +  2 * num_limbs * (num_add(L)+1) * (num_add(R) + 1)),
     //      num_groups = Ceil[num_limbs / S],
     // ``
```
Comment (DDT92) [inline r1cs/gadgets/std/src/fields/nonnative/nonnative_field_gadget.rs:1117]: num_limbs is the number of limbs of a NonNativeFieldGadget. A NonNativeFieldMulResultGadget has 2 * num_limbs - 1  limbs. So we should have
num_groups = Ceil[(2*num_limbs - 1) / S].

File: r1cs/gadgets/std/src/fields/nonnative/nonnative_field_mul_result_gadget.rs [hunk 3]
```diff
@@ -328,20 +334,20 @@ impl<SimulationF: PrimeField, ConstraintF: PrimeField>
...
+        //                      <= (num_limbs + 2*(num_add + 1)) * 2^{2*bits_per_limb[i]}. 
         // ``
         // Hence we may set num_adds and surfeit for the limbs of `k*p + r` according
         // to
         // ``
-        //      num_adds(kp + r) = num_limbs + num_add,
-        //      surfeit(kp + r) = log(num_limbs + num_adds + 1).
+        //      num_adds(kp + r) = num_limbs + 2*num_adds + 1,
+        //      surfeit(kp + r) = log(num_limbs + 2*(num_adds + 1)).
         // ``
```
Comment (DDT92) [inline r1cs/gadgets/std/src/fields/nonnative/nonnative_field_mul_result_gadget.rs:346]: I remember I suggested surfeit(kp + r) = len( num_limbs + 2^len(num_adds + 1)). This overestimates it a little bit in some situations, but it is still fine.

**Review by DDT92 — APPROVED** [review]
Everything looks correct after the  last changes.

**Pull request merged**